internal/pkg/install/prepare.go

```diff
@@ -33,281 +62,260 @@ func Prepare(data *userdata.UserData) (err error) {
-		data.Install.Data.Size = 1024 * 1000 * 1000
+	manifest := NewManifest(data)
+
+	if data.Install.Wipe {
+		if err = WipeDevices(manifest); err != nil {
+			return errors.Wrap(err, "failed to wipe install device(s)")
+		}
 	}
 
-	// Boot Device Init
-	if data.Install.Boot != nil {
-		if data.Install.Boot.Device == "" {
-			data.Install.Boot.Device = data.Install.Root.Device
+	// Create and format all partitions.
+
+	for dev, targets := range manifest.Targets {
+		var bd *blockdevice.BlockDevice
+		if bd, err = blockdevice.Open(dev, blockdevice.WithNewGPT(data.Install.Force)); err != nil {
+			return err
 		}
-		if data.Install.Boot.Size == 0 {
-			// Set to 512MB default for funzies
-			data.Install.Boot.Size = 512 * 1000 * 1000
+		// nolint: errcheck
+		defer bd.Close()
+
+		for _, target := range targets {
+			if err = target.Partition(bd); err != nil {
+				return errors.Wrap(err, "failed to partition device")
+			}
 		}
-		if len(data.Install.Boot.Data) == 0 {
-			data.Install.Boot.Data = append(data.Install.Boot.Data, "https://github.com/autonomy/talos/releases/download/"+version.Tag+"/vmlinuz")
-			data.Install.Boot.Data = append(data.Install.Boot.Data, "https://github.com/autonomy/talos/releases/download/"+version.Tag+"/initramfs.xz")
 
+		if err = bd.RereadPartitionTable(); err != nil {
+			return err
 		}
-	}
 
-	// Verify that the disks are unused
-	// Maybe a simple check against bd.UUID is more appropriate?
-	if !data.Install.Wipe {
-		var dev *probe.ProbedBlockDevice
-		for _, device := range []string{data.Install.Boot.Device, data.Install.Root.Device, data.Install.Data.Device} {
-			dev, err = probe.GetDevWithFileSystemLabel(device)
-			if err != nil {
-				// We continue here because we only care if we can discover the
-				// device successfully and confirm that the disk is not in use.
-				// TODO(andrewrynhard): We should return a custom error type here
-				// that we can use to confirm the device was not found.
-				continue
-			}
-			if dev.SuperBlock != nil {
-				return errors.Errorf("target install device %s is not empty, found existing %s file system", device, dev.SuperBlock.Type())
+		for _, target := range targets {
+			if err = target.Format(); err != nil {
+				return errors.Wrap(err, "failed to format device")
 			}
 		}
 	}
 
-	// Create a map of all the devices we need to be concerned with
-	devices := make(map[string]*Device)
-	labeldev := make(map[string]string)
+	return err
+}
 
-	// PR: Should we only allow boot device creation if data.Install.Wipe?
-	if data.Install.Boot.Device != "" {
-		devices[constants.BootPartitionLabel] = NewDevice(data.Install.Boot.Device,
-			constants.BootPartitionLabel,
-			data.Install.Boot.Size,
-			data.Install.Wipe,
-			false,
-			data.Install.Boot.Data)
-		labeldev[constants.BootPartitionLabel] = data.Install.Boot.Device
+// VerifyRootDevice verifies the supplied root device options.
+func VerifyRootDevice(data *userdata.UserData) (err error) {
```

Comment:
I wonder if these 'default' settings should be added to user data instead?

Reply:
I think so eventually. We should add it in when we do #268.